### DISCUSSION TITLE
Add option for a "forced" kiosk mode

### DISCUSSION THREAD

**Discussion by shaharmor** [grafana/grafana#51295] [Dashboards, Panels & Visualizations]
Hi,

We are using Grafana as a dashboard for users with limited access, which should only be able to view the dashboard that we provide them (Without all the additional UI elements like the sidebar, choosing additional dashboards, etc).

We are using the Kiosk mode for that which is working well, but we are facing one issue:
When using Kiosk mode, if the user clicks "ESC", they exit the Kiosk mode, which then updates the UI with all the stuff that we are trying to hide from the user.
We are looking for a way to "force" kiosk mode without proving any option to exit it, or even without any option to cycle between the different Kiosk modes.

Would it be possible to add a global configuration to force kiosk mode to a specific mode without any option to exit that mode?

**Comment by acontreras89** [comment]
We are having this very same need when embedding dashboards through iframes in external (web) applications. Being able to move around Grafana leads to a very poor user experience.

**Comment by ryanwuer** [comment]
Looking forward to this feature.

**Comment by bysph** [comment]
Looking forward to this feature.

**Comment by pmint93** [comment]
In the meantime, I have a fork that added the `kiosk=embedded` mode with following changes: hide the sidebar, toolbar (but keep the timepicker), cannot switch to another kiosk mode or escape from it. However, I'm not sure this is the right approach.

https://github.com/pmint93/grafana/tree/embedded-kiosk-mode

Back-ported to:

v9.0.4 https://github.com/pmint93/grafana/commits/v9.0.4-embedded-kiosk-mode

v8.2.7 https://github.com/pmint93/grafana/tree/v8.2.7-embedded-kiosk-mode

Demo:

<img width="808" alt="image" src="https://user-images.githubusercontent.com/7886909/192205410-784aa299-7da7-4af4-b6aa-96c02369f558.png">

**Comment by aaronwjha** [comment]
This feature would be very helpful. Any news on getting it added?

**Comment by natellium** [comment]
👋 this use case seems more suited for [Public Dashboards](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/) - they have some limitations in terms of variables support, datasources, etc. but that's what the feature is intended for.

**Reply by sharifanani** [reply]
I love Public Dashboards, but without supporting template variables, I've needed to hack around kiosk mode & proxying etc etc :(
